src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -563,8 +563,10 @@ public static void Move(String sourceDirName, String destDirName)
             String destinationRoot = Path.GetPathRoot(destPath);
             if (!String.Equals(sourceRoot, destinationRoot, pathComparison))
                 throw new IOException(SR.IO_SourceDestMustHaveSameRoot);
-            
-            if (!FileSystem.Current.DirectoryExists(fullsourceDirName))
+
+            // Windows will throw if the source file/directory doesn't exist, we preemptively check
+            // to make sure our cross platform behavior matches NetFX behavior.
+            if (!FileSystem.Current.DirectoryExists(fullsourceDirName) && !FileSystem.Current.FileExists(fullsourceDirName))
```

Comment:
It's unfortunate we're now making multiple calls to FillAttributeInfo on Windows and FileExists on Unix: each one appears to do the same thing but then checks whether the result is a directory or a file.  Should we add a DirectoryOrFileExists to the pal so that the implementation can do these checks once rather than twice?